From 18f8405807fa79f50cae15787698778736df8c9b Mon Sep 17 00:00:00 2001
From: AndreaM12345 <ayma@us.ibm.com>
Date: Wed, 18 May 2022 14:32:42 -0500
Subject: remove warnings from istioctl envoy filter analyzer during first
 install (#38998)

* set priority in default envoyFilters to -1

* update release note msg

* update test
---
 .../istio-control/istio-discovery/files/gen-istio.yaml | 10 ++++++++++
 .../istio-discovery/templates/telemetryv2_1.11.yaml    |  5 +++++
 .../istio-discovery/templates/telemetryv2_1.12.yaml    |  5 +++++
 .../istio-discovery/templates/telemetryv2_1.13.yaml    |  5 +++++
 .../istio-discovery/templates/telemetryv2_1.14.yaml    |  5 +++++
 .../istio-discovery/templates/telemetryv2_1.15.yaml    |  5 +++++
 .../istiod-remote/templates/telemetryv2_1.11.yaml      |  5 +++++
 .../istiod-remote/templates/telemetryv2_1.12.yaml      |  5 +++++
 .../istiod-remote/templates/telemetryv2_1.13.yaml      |  5 +++++
 .../istiod-remote/templates/telemetryv2_1.14.yaml      |  5 +++++
 .../istiod-remote/templates/telemetryv2_1.15.yaml      |  5 +++++
 releasenotes/notes/38676.yaml                          |  8 ++++++++
 12 files changed, 68 insertions(+)
 create mode 100644 releasenotes/notes/38676.yaml

diff --git a/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml b/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml
index 5e67a12e1e..a305453c46 100644
--- a/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml
+++ b/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml
@@ -1782,6 +1782,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -1906,6 +1907,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -2022,6 +2024,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -2146,6 +2149,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -2262,6 +2266,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -2386,6 +2391,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -2502,6 +2508,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -2626,6 +2633,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -2742,6 +2750,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -2866,6 +2875,7 @@ metadata:
   labels:
     istio.io/rev: default
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml
index fba3a5ec22..ba45e1519f 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -451,6 +454,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -564,6 +568,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml
index aeb987f23e..4db6db7ed1 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -451,6 +454,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -564,6 +568,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.13.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.13.yaml
index ae5f5c2cb2..c7cbe868a5 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.13.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.13.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -463,6 +466,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -586,6 +590,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.14.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.14.yaml
index ccef10d25d..5ecfead2be 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.14.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.14.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -463,6 +466,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -586,6 +590,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.15.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.15.yaml
index baf6080b79..b9d9dc5af2 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.15.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.15.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -463,6 +466,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -586,6 +590,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml
index fba3a5ec22..ba45e1519f 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -451,6 +454,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -564,6 +568,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml
index aeb987f23e..4db6db7ed1 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -451,6 +454,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -564,6 +568,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.13.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.13.yaml
index ae5f5c2cb2..c7cbe868a5 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.13.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.13.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -463,6 +466,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -586,6 +590,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.14.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.14.yaml
index ccef10d25d..5ecfead2be 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.14.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.14.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -463,6 +466,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -586,6 +590,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.15.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.15.yaml
index baf6080b79..b9d9dc5af2 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.15.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.15.yaml
@@ -14,6 +14,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
@@ -177,6 +178,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: NETWORK_FILTER
       match:
@@ -333,6 +335,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
 {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
     - applyTo: HTTP_FILTER
@@ -463,6 +466,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
   {{- if not .Values.telemetry.v2.stackdriver.disableOutbound }}
   - applyTo: NETWORK_FILTER
@@ -586,6 +590,7 @@ metadata:
   labels:
     istio.io/rev: {{ .Values.revision | default "default" }}
 spec:
+  priority: -1
   configPatches:
     - applyTo: HTTP_FILTER
       match:
diff --git a/releasenotes/notes/38676.yaml b/releasenotes/notes/38676.yaml
new file mode 100644
index 0000000000..8256428233
--- /dev/null
+++ b/releasenotes/notes/38676.yaml
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: installation
+issue:
+  - 38676
+releaseNotes:
+  - |
+    **Fixed** change to add priority of -1 to envoyFilters deployed by default by istio to remove warnings from istioctl envoyFilter analyzer on first install
-- 
2.35.3

